src/main/java/org/terracotta/offheapstore/disk/storage/FileBackedStorageEngine.java

```diff
@@ -624,41 +552,34 @@ Long writeMappingBuffers(ByteBuffer keyBuffer, ByteBuffer valueBuffer, int pojoH
     }
 
     WriteContext getKeyWriteContext(long address) {
-      try {
-        long position = filePosition + address;
-        FileWriteTask pending = pendingWrites.get(position);
-        if (pending == null) {
-          int keyLength = readIntFromChannel(position + KEY_LENGTH_OFFSET);
-          return getDiskWriteContext(position + KEY_DATA_OFFSET, keyLength);
+      long position = filePosition + address;
+      FileWriteTask pending = pendingWrites.get(position);
+      if (pending == null) {
+        int keyLength = readBuffer(position + KEY_LENGTH_OFFSET, 4).getInt(0);
+        return getDiskWriteContext(position + KEY_DATA_OFFSET, keyLength);
+      } else {
+        if (pendingWrites.get(position) != pending) {
+          return getKeyWriteContext(address);
         } else {
-          if (pendingWrites.get(position) != pending) {
-            return getKeyWriteContext(address);
-          } else {
-            return getQueuedWriteContext(pending, pending.getKeyBuffer());
-          }
+          return getQueuedWriteContext(pending, pending.getKeyBuffer());
         }
-      } catch (IOException e) {
-        throw new RuntimeException(e);
       }
     }
 
     WriteContext getValueWriteContext(long address) {
-      try {
-        long position = filePosition + address;
-        FileWriteTask pending = pendingWrites.get(position);
-        if (pending == null) {
-          int keyLength = readIntFromChannel(position + KEY_LENGTH_OFFSET);
-          int valueLength = readIntFromChannel(position + VALUE_LENGTH_OFFSET);
-          return getDiskWriteContext(position + keyLength + KEY_DATA_OFFSET, valueLength);
+      long position = filePosition + address;
+      FileWriteTask pending = pendingWrites.get(position);
+      if (pending == null) {
+        ByteBuffer lengths = readBuffer(position + KEY_LENGTH_OFFSET, 8);
+        int keyLength = lengths.getInt(0);
+        int valueLength = lengths.getInt(VALUE_LENGTH_OFFSET - KEY_LENGTH_OFFSET);
+        return getDiskWriteContext(position + keyLength + KEY_DATA_OFFSET, valueLength);
```

Comment:
Same as the comment on `readValueBuffer` above.

Reply:
More rebasing woes.